addons/account/models/account_journal_dashboard.py

```diff
@@ -1019,10 +1016,11 @@ def open_action(self):
 
         context = self.env.context.copy()
         if 'context' in action and isinstance(action['context'], str):
-            context.update(ast.literal_eval(action['context']))
+            action_context = ast.literal_eval(action['context'])
         else:
-            context.update(action.get('context', {}))
-        action['context'] = context
+            action_context = action.get('context', {})
+        action_context.update(context)
+        action['context'] = action_context
         action['context'].update({
             'default_journal_id': self.id,
         })
```

Comment:
What you did is fine. But since we are on `master` I think sth. like this would work and be a bit cleaner
```suggestion
        action['context'] = {
            **action_context,
            **self.env.context,
            'default_journal_id': self.id,
        }
```
(and remove the `context` variable)